mava/components/jax/training/model_updating.py

```diff
@@ -93,6 +98,19 @@ def on_training_utility_fns(self, trainer: SystemTrainer) -> None:
             None.
         """
 
+        # Initilaise running statisitics here
+
+        if self.config.normalize_target_values:
+            trainer.store.running_stats_fn = compute_running_mean_var_count
+            running_stats = jnp.array([0, 0, 1e-4])
+        else:
+            running_stats = jnp.array([0, 1, 1e-4])
+            trainer.store.running_stats_fn = dummy_running_mean_var_count
+
+        trainer.store.stats = {}
+        for net_key in trainer.store.trainer_agent_net_keys.keys():
+            trainer.store.stats[net_key] = running_stats
```

Comment:
Do we capture stats other than the running stats here? Because if it is just the running stats, we could perhaps create the empty stats dict inside the if statement, remove the else statement and not need a dummy stats function? Then we only call the function from the store if it exists using something like `if hasattr(trainer.store, 'stats'): ...` In fact, this might be a recurring thing that happens across Mava, i.e. the store might have something or not, based on a boolean variable in the config of a component. So perhaps we can create a function for the store to check, e.g. `store.has("name_of_attribute")` which just returns `hasattr(self, "name_of_attribute")` to make checks like these neater. Let me know what you think. Also @KaleabTessera and @DriesSmit, your opinions as well 🙂

Reply:
Yes I think that's a good idea to have a function that checks store attributes. I was trying to keep the code clean with the dummy function to avoid if statements later in the sgd_step function. But if the dummy function is confusing we can remove that and follow your suggestion. The running stats are initialised differently. The dummy ones that never change have variance of 1. In the other case the initial value is 0 so we will still need an else statement to capture the difference.

Reply:
@arnupretorius I removed the unnecessary dummy function.

Reply:
Thanks @ulricharmel , will take a look :)